### ISSUE TITLE
Recipe: qwen3.5-0.8b-grpo and qwen3.5-2b-grpo, the small-model reasoning variants (part of #275)

### ISSUE THREAD

**Issue by MakazhanAlpamys** [MakazhanAlpamys/Soup#848]
## What

Add two recipes to `src/soup_cli/recipes/catalog.py`:

- `qwen3.5-0.8b-grpo`
- `qwen3.5-2b-grpo`

`Qwen/Qwen3.5-0.8B` and `Qwen/Qwen3.5-2B` ship SFT only. Unlike most #275 rows, these two are small enough that a contributor might actually run a few steps of them — which makes them the most useful rows in the tracker for anyone learning GRPO on their own hardware. **Comment on this issue before you start.**

## Base repos

Both checked on 2026-09-11 with `huggingface_hub.model_info(..., token=False)`: they resolve, are not gated, and report `model_type: qwen3_5`. Re-check them the same way; an HTTP status is not evidence (#661).

## What to copy, and from where

- Siblings: `qwen3.5-0.8b-sft` (`lr: 3e-4`, LoRA r=8 / alpha=16, `max_length: 2048`) and `qwen3.5-2b-sft` (`lr: 2e-4`, same LoRA and length). Both carry `modality: text`, `quantization: 4bit`, `batch_size: auto`.
- Task block: `r1-distill-qwen-1.5b-grpo`, the only small (1.5B) Qwen-architecture GRPO recipe, or `qwen3.5-9b-grpo` for the in-family choice. They agree on `lr: 1e-5`, `grpo_beta: 0.1`, `num_generations: 4`, `reward_fn: accuracy`, `batch_size: auto`, `gradient_accumulation_steps: 8`, LoRA r=16.

Decisions you will need to make and state, not inherit silently:

- **Learning rate.** The SFT siblings use 2e-4 / 3e-4; across all 23 GRPO recipes the values are `1e-5` (16), `5e-6` (4) and `1e-6` (3, the 135M `grpo-env-*` recipes). Copying the SFT rate into a GRPO recipe would be an outlier by more than an order of magnitude; say which GRPO precedent you follow.
- **LoRA rank.** The siblings use r=8, the GRPO templates r=16. Either is defensible; pick one and say why.
- **`max_length`.** The siblings use 2048 and the GRPO templates 4096; reasoning completions are long, so this matters more here than for SFT.

YAML assembled from the sibling plus the GRPO block parses through `load_config_from_string` on current `main`. That proves the schema accepts it, nothing more.

## The ratchets that must pass

- [ ] `RecipeMeta(...)` entries, tags following the siblings plus `grpo` / `reasoning`.
- [ ] `modality: text` in both YAMLs and both names added to `EXPECTED_QWEN35_TEXT_RECIPES` in `tests/test_issue427_qwen35_text_modality.py:11`; the family audit at `:43` fails otherwise (#427).
- [ ] Model id pinned on **two independent surfaces** (`RecipeMeta.model` and YAML `base:`), each by its own test — mirror `TestDeepSeekV4FlashDpoRecipe`, `tests/test_recipes.py:503`.
- [ ] Catalog count bumped everywhere (165 today; use the number at rebase time): `catalog.py:51`; `test_catalog_size_is_N` in `tests/test_recipes.py:936`, `tests/test_v07130.py:740`, `tests/test_v07132.py:953` plus the history line near `tests/test_recipes.py:964`; `tests/test_v07124.py:281` and the docstring at `:277`; the doc sites in `tests/test_recipe_count_is_synced.py:83-96`.
- [ ] `python scripts/generate_recipe_snapshot.py` run last; additive diff only.
- [ ] Changelog fragment `changelog.d/0.74.0/<PR>.added.md`.
- [ ] `soup recipes show` / `use` run live for both; paste the output.

## Optional, and welcome

If you have a GPU, a short `soup train` smoke run (a handful of steps on a small reasoning set) with the card, VRAM and whether the loss moved stated in the PR would make this the first #275 recipe with any runtime evidence. It is **not** required for merge, and please do not present a few steps as a tuned recommendation.

## A note on mutations

`grpo_beta: 0.1`, `num_generations: 4`, `reward_fn: accuracy`, `epochs: 3`, `batch_size: auto` equal their schema defaults; deleting them is an equivalent mutation. Report them as survivors.

**Comment by SID-6921** [comment]
Taking this one. Plan: base LR on the GRPO precedent (1e-5, the majority convention across the 23 existing GRPO recipes) rather than the SFT siblings' 2e-4/3e-4, LoRA r=16 matching the GRPO templates, max_length: 4096 given reasoning completions run long. Will state each choice explicitly in the PR rather than silently inheriting from either side, re-verify both base repos live, and cover every ratchet listed (RecipeMeta tags, EXPECTED_QWEN35_TEXT_RECIPES, the two-independent-surface model-id pin test, every catalog-count site, regenerated snapshot, changelog fragment, live soup recipes show/use output).

**Comment by MakazhanAlpamys** [comment]
Fixed in `f454bfe4`, by **@SID-6921** in #864. `main` is now at 167 recipes.

Both recipes reuse the `qwen3.5-9b-grpo` block byte-for-byte except the `base:` line — the in-family template this issue offered — and the three values the issue asked to be *stated rather than inherited* are stated: `lr: 1e-5` (16 of 23 existing GRPO recipes; the SFT siblings' 2e-4/3e-4 would be an order of magnitude out on a GRPO objective), LoRA `r=16/alpha=32`, and `max_length: 4096`.

Repo ids verified live against the Hub with the same anonymous `model_info(..., token=False)` call `scripts/check_recipe_repo_ids.py` makes — `Qwen/Qwen3.5-0.8B` and `Qwen/Qwen3.5-2B` both resolve, neither gated nor private. Worth recording why that was done by hand: `recipe-repo-ids.yml` is schedule-only with no `pull_request` trigger by design, and the in-repo guard test uses a fake Hub client, so a PR gets no live id check otherwise.

Both carry the two-surface literal repo-id pin, and it was measured to be load-bearing: a consistently-wrong id — wrong in `RecipeMeta.model` **and** the inline `base:`, so the two still agree — leaves every catalog-wide cross-check silent and is caught only by those pins. That is the `glm-5`/THUDM class.

Try them:

```
soup recipes show qwen3.5-2b-grpo
soup recipes use qwen3.5-0.8b-grpo
```

Neither was executed — the hyperparameters come from the in-family template rather than a tuning run, as this issue asked.

**One correction for whoever picks up the next part of #275:** the issue's note on equivalent-by-construction mutations lists five default-equal fields; there is a sixth. `TrainingConfig.quantization` defaults to `'4bit'`, so `quantization: 4bit` is also equivalent by construction and its deletion survives — measured during this review.
